haystack/components/builders/chat_prompt_builder.py

```diff
@@ -125,22 +131,25 @@ def __init__(
         self.required_variables = required_variables or []
         self.template = template
         variables = variables or []
-        try:
-            # The Jinja2TimeExtension needs an optional dependency to be installed.
-            # If it's not available we can do without it and use the ChatPromptBuilder as is.
-            self._env = SandboxedEnvironment(extensions=[Jinja2TimeExtension])
-        except ImportError:
-            self._env = SandboxedEnvironment()
+
+        self._env = SandboxedEnvironment(extensions=[ChatMessageExtension])
```

Comment:
Oh cool! Is this also how we could have overcome this issue https://github.com/deepset-ai/haystack/issues/8517?

Reply:
At one point, I even tried to think about #8517, without much success.

My current understanding is that if we want Jinja to return objects, we should use `NativeEnvironment`, which exposes us to risks.

What we can do instead and I am doing here: have Jinja return a JSON string, while the component does the deserialization.

Reply:
Ahh okay I see. Let's leave it alone then for now.